core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -188,7 +224,71 @@ def start_board(self, boards: List[Tuple[FlightControllerType, str]]) -> bool:
         raise RuntimeError("Invalid board type: {boards}")
 
     def restart(self) -> None:
-        self.mavlink_manager.restart()
+        # Ensures all supported run methods restart properly by stopping Mavlink Manager and restarting main loops
+        self.mavlink_manager.stop()
+        if self.current_platform != Platform.Undefined:
+            if self.current_platform == Platform.SITL:
+                self.run_with_sitl(self.current_sitl_frame)
+                return
+        self.run_with_board()
+
+    def is_vehicle_armed(self) -> bool:
+        get_response = self.messenger.get_mavlink_message("HEARTBEAT")
+        base_mode_bits = get_response["message"]["base_mode"]["bits"]
+        if not isinstance(base_mode_bits, int):
+            raise ValueError("Got unexpected HEARTBEAT message from Autopilot.")
+        return base_mode_bits >= 128
```

Comment:
I don't like this. It works for now because MAV_MODE_FLAG_SAFETY_ARMED is the last flag in the array, but I would rather see 
```suggestion
        return bool(base_mode_bits & MAV_MODE_FLAG_SAFETY_ARMED)
```

Reply:
Yeah. I was sloppy on this one hehe

Changed for a bit check:

```python
return bool(bin(base_mode_bits)[-8])
```

Reply:
Ok, got your suggestion now. Will take bitwise operator to life.